source/rust_verify_test/tests/syntax_attr.rs

```diff
@@ -670,3 +670,25 @@ test_verify_one_file! {
 
     } => Err(e) => assert_rust_error_msg(e, "method `spec_f` is not a member of trait `X`")
 }
+
+test_verify_one_file! {
+    #[test] test_macro_inside_proof_decl code!{
+        macro_rules! inline_proof {
+            ($val: expr => $x: ident, $y: ident) => {
+                proof_decl!{
+                    let tracked $x: int = $val;
+                    let ghost $y: int = $val;
+                    assert($x == $y);
+                }
+             }
+        }
+
+        fn test_inline_proof_via_macro() {
+            proof_decl!{
+                inline_proof!(0 => x, y);
```

Comment:
Here, `inline_proof!` expands to a `proof_decl!`, but this already inside another `proof_decl!`.  Is it intended that you need to nest `proof_decl!`s inside each other like this?

Reply:
It is intended. I want to use macro inside `proof_decl!{}` so that it is clear that the new macro is a proof instead of an arbitrary macro used in execution. Unfortunately, `proof_decl!` does not see content expanded from `inline_proof!`, so I need to explicitly call `proof_decl!` or `proof!` in macro definition.